shared/manifest.go

```diff
@@ -0,0 +1,53 @@
+// Copyright 2018 The WPT Dashboard Project. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package shared
+
+import "strings"
+
+// explosions returns a map of the exploded test by filename suffix.
+func explosions() map[string][]string {
+	return map[string][]string{
+		".window.js": []string{".window.html"},
+		".worker.js": []string{".worker.html"},
+		".any.js": []string{
+			".any.html",
+			".any.worker.html",
+			".any.serviceworker.html",
+			".any.sharedworker.html",
+		},
+	}
+}
+
+// ExplodePossibleRenames returns a map of equivalent renames for the given file rename.
+func ExplodePossibleRenames(before, after string) map[string]string {
+	result := map[string]string{
+		before: after,
+	}
+	eBefore := ExplodePossibleFilenames(before)
+	eAfter := ExplodePossibleFilenames(after)
+	if len(eBefore) == len(eAfter) {
```

Comment:
Optional: we could use more heuristics here if the lengths are different.

e.g. `.any.html` can be compared against `.window.html`, and `.any.worker.html` can be compared to `.worker.html`.

Reply:
Might follow up with other cases.